### DISCUSSION TITLE
Semver en changelog

### DISCUSSION THREAD

**Discussion by sanderdekroon** [OpenWebconcept/Afspraken#4] [General]
Ik wil voorstellen om semantic versioning toe te passen voor alle plugins binnen het Open Webconcept. Zo maken we direct duidelijk of een update eventueel invloed kan hebben op gemaakt maatwerk. Zie https://semver.org/ voor inhoudelijke informatie. 

De TL;DR is:
- Een versie bestaat uit {MAJOR}.{MINOR}.{PATCH}
- Een major version betekent dat er een _backward incompatible_ verandering wordt aangebracht in de publieke API (code + json api?)
- Een minor version betekent nieuwe functionaliteit(en) die _backward compatible_ is geïntegreerd
- Een patch version betekent bug fix(es) die _backward compatible_ is geïntegreerd

Daarnaast wil ik voorstellen om een CHANGELOG.MD bestand bij te gaan houden in elke repository op basis van het [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

**Comment by MWest2020** [comment]
Ik hoop dat iedereen het hier mee eens gaat zijn. Omdat we te maken hebben met meerdere partijen is het fijn om te weten wat er precies gebeurt bij het bijwerken van de plug-ins.

Staat het al op het board om besproken te worden?

**Comment by Htbaa** [comment]
Goed plan. Ook groot fan van een goede changelog. Voor DataInMap ook al vanaf de publieke release bijgehouden. Wel in een iets ander formaat, maar het opdelen in de categorieën added/changed/removed/etc. kan ik waarderen. In ieder geval géén CHANGELOG genereren op de commit history, dat voegt doorgaans niks toe.
